devito/finite_differences/derivative.py

```diff
@@ -192,22 +199,39 @@ def T(self):
             adjoint = direct
 
         return Derivative(self.expr, *self.dims, deriv_order=self.deriv_order,
-                          fd_order=self.fd_order, side=self.side, stagger=self.stagger,
-                          transpose=adjoint)
+                          fd_order=self.fd_order, side=self.side, transpose=adjoint,
+                          x0=self.x0, subs=self._subs)
+
+    def _eval_at(self, func):
+        """
+        Evaluates the derivative at the location of `func`. It is necessary for staggered
+        setup where one could have Eq(u(x + h_x/2), v(x).dx)) in which case v(x).dx
+        has to be computed at x=x + h_x/2.
+        """
+        x0 = {d1: d2 for d1, d2 in zip(func.dimensions, func.indices_ref)}
+        return Derivative(self.expr, *self.dims, deriv_order=self.deriv_order,
+                          fd_order=self.fd_order, side=self.side,
+                          transpose=self.transpose, subs=self._subs, x0=x0)
 
     @property
     def evaluate(self):
-        expr = getattr(self.expr, 'evaluate', self.expr)
+        # Evaluate finite-difference.
+        # Note: evaluate and _eval_fd splitted for potential future different
+        # types of discretizations.
+        return self._eval_fd(self.expr)
```

Comment:
_eval_fd now appears to be useless. Squash it within evaluate ?

Reply:
" Note: evaluate and _eval_fd splitted for potential future different types of discretizations."

That's literally the line above